includes/llms.functions.core.php

```diff
@@ -367,42 +367,29 @@ function llms_get_date_diff( $time1, $time2, $precision = 2 ) {
  * This function suppresses PHP warnings that would be thrown by DOMDocument when
  * loading a partial string or an HTML string with errors.
  *
+ * @see LLMS_DOM_Document->load().
+ *
  * @since 4.7.0
  * @since 4.8.0 Remove reliance on `mb_convert_encoding()`.
+ * @since [version] Add back partial reliance on `mb_convert_encoding()` but keep the previous implementation as a fall-back.
+ *               Also fix a potential fatal in the fall-back which tried to manipulate a non existent node.
+ *               Wrapper for `LLMS_Dom_Document:load()`.
  *
  * @param string $string An HTML string, either a full HTML document or a partial string.
  * @return DOMDocument|WP_Error Returns an instance of DOMDocument with `$string` loaded into it
  *                              or an error object when DOMDocument isn't available or an error is encountered during loading.
  */
 function llms_get_dom_document( $string ) {
 
-	if ( ! class_exists( 'DOMDocument' ) ) {
-		return new WP_Error( 'llms-dom-document-missing', __( 'DOMDocument not available.', 'lifterlms' ) );
-	}
-
-	// Don't throw or log warnings.
-	$libxml_state = libxml_use_internal_errors( true );
-
-	// This forces DOMDocument to convert non-utf8 characters into HTML entities and without relying on `mb_convert_encoding()`.
-	$utf8_fixer = '<meta id="llms-get-dom-doc-utf-fixer" http-equiv="Content-Type" content="text/html; charset=utf-8">';
+	$llms_dom = new LLMS_DOM_Document( $string );
 
-	$dom = new DOMDocument();
-	if ( ! $dom->loadHTML( $utf8_fixer . $string ) ) {
-		$dom = new WP_Error( 'llms-dom-document-error', __( 'DOMDocument XML Error encountered.', 'lifterlms' ), libxml_get_errors() );
+	if ( is_wp_error( $llms_dom ) ) {
```

Comment:
`$llms_dom` will never be a `WP_Error` because the constructor won't ever return anything as noted in the note on the class constructor.

So if we do encounter this error we'll proceed anyway and I think we'll have weird unexpected results.